rollbar/__init__.py

```diff
@@ -86,17 +86,68 @@
 except ImportError:
     AppEngineFetch = None
 
+def passthrough_decorator(func):
+    def wrap(*args, **kwargs):
+        return func(*args, **kwargs)
+    return wrap
+
 try:
     from tornado.gen import coroutine as tornado_coroutine
     from tornado.httpclient import AsyncHTTPClient as TornadoAsyncHTTPClient
 except ImportError:
-    def tornado_coroutine(func):
-        def wrap(*args, **kwargs):
-            return func(*args, **kwargs)
-        return wrap
-
+    tornado_coroutine = passthrough_decorator
     TornadoAsyncHTTPClient = None
 
+try:
+    from twisted.internet import reactor
+    from twisted.web.client import Agent as TwistedHTTPClient
+    from twisted.internet.defer import inlineCallbacks, Deferred, returnValue
+    from twisted.internet.protocol import Protocol
+    from twisted.web.http_headers import Headers as TwistedHeaders
+    from zope.interface import implements
+
+    from twisted.internet.defer import succeed
+    from twisted.web.iweb import IBodyProducer
+
+
+    class StringProducer(object):
+        implements(IBodyProducer)
+
+        def __init__(self, body):
+            self.body = body
+            self.length = len(body)
+
+        def startProducing(self, consumer):
+            consumer.write(self.body)
+            return succeed(None)
+
+        def pauseProducing(self):
+            pass
+
+        def stopProducing(self):
+            pass
+
+
+    class ResponseAccumulator(Protocol):
+        def __init__(self, length, finished):
+            self.remaining = length
+            self.finished = finished
+            self.response = ''
+
+        def dataReceived(self, bytes):
+            if self.remaining:
+                chunk = bytes[:self.remaining]
+                self.response += chunk
+                self.remaining -= len(chunk)
+
+        def connectionLost(self, reason):
+            self.finished.callback(self.response)
```

Comment:
Since Rollbar requires an HTTPS connection, I would add in the following check for OpenSSL:

```py
try:
    # Verify we can make HTTPS requests with Twisted.
    # From http://twistedmatrix.com/documents/12.0.0/core/howto/ssl.html
    from OpenSSL import SSL
except ImportError:
    log.exception('Rollbar requires SSL to work with Twisted')
    raise 
```